server/server.go

```diff
@@ -986,6 +990,27 @@ func (s *Server) SetAuditConfig(cfg config.AuditConfig) error {
 	return nil
 }
 
+// GetRateLimitConfig gets the rate limit config information.
+func (s *Server) GetRateLimitConfig() *config.RateLimitConfig {
+	return s.serviceMiddlewarePersistOptions.GetRateLimitConfig().Clone()
+}
+
+// SetRateLimitConfig sets the rate limit config.
+func (s *Server) SetRateLimitConfig(cfg config.RateLimitConfig) error {
+	old := s.serviceMiddlewarePersistOptions.GetRateLimitConfig()
+	s.serviceMiddlewarePersistOptions.SetRateLimitConfig(&cfg)
+	if err := s.serviceMiddlewarePersistOptions.Persist(s.storage); err != nil {
```

Comment:
Oh, Should we make this function threadsafe? Was it possible that two goroutines run this function, for example, should we avoid the following case:
1. go1 finished L1001 and update the cfg to false
2. go2 finished 1001 and update the cfg to true in cache
3. go2 persist the value with true (from cache) and return success
4. go1 persist the value with true(from cache) while it expect the value is false.
and more different cases may meet in the `Persist` with it not protect while read and write..

Reply:
This happens to all config that needs to `Persist`, and if optimization is needed, I think we can fix it together in another PR